app/src/main/java/com/asfoundation/wallet/viewmodel/TransactionsViewModel.java

```diff
@@ -406,6 +407,33 @@ public void showSettings(Context context) {
     transactionViewNavigator.openSettings(context, false);
   }
 
+  public void goToVipLink(Context context) {
+    Uri uri =
+        Uri.parse("https://appcoins.medium.com/welcome-to-the-appcoins-vip-program-7e5d133cc0ad");
+    transactionViewNavigator.navigateToBrowser(context, uri);
+  }
+
+  public MutableLiveData<Boolean> shouldShowVipBadge() {
+    return showVipBadge;
+  }
+
+  public void verifyUserLevel() {
+    disposables.add(transactionViewInteractor.findWallet()
+        .subscribeOn(networkScheduler)
+        .flatMap(wallet -> transactionViewInteractor.getUserLevel()
+            .subscribeOn(networkScheduler)
+            .map(userLevel -> {
```

Comment:
Map seems a little bit ill-used, you could just use `doOnSuccess` instead and straight up use `showVipBadge.postValue(userLevel== 9 || userLevel== 10);` instead of using indirection by calling a method (but it's not important).